nexus/external-api/src/v2025112000.rs

```diff
@@ -62,29 +56,36 @@ impl TryFrom<external::Disk> for Disk {
     type Error = dropshot::HttpError;
 
     fn try_from(new: external::Disk) -> Result<Disk, Self::Error> {
+        let (snapshot_id, image_id) = match new.disk_type {
+            external::DiskType::Distributed { snapshot_id, image_id } => {
+                (snapshot_id, image_id)
+            }
+
+            _ => {
+                // Cannot display any other variant for this old client
+                return Err(dropshot::HttpError::for_client_error(
+                    Some(String::from("Not Acceptable")),
+                    dropshot::ClientErrorStatusCode::NOT_ACCEPTABLE,
+                    String::from(
+                        "disk type variant not supported for client version",
+                    ),
+                ));
+            }
+        };
+
+        // This path was fake before, so retain that behaviour here
+        let device_path = format!("/mnt/{}", new.identity.name.as_str());
```

Comment:
why do we return fake data? if we're doing this change, can we remove this field if it's useless?

Reply:
Unfortunately we have to maintain compatibility with the previous API versions, in case anyone has cached this result and will compare against it.